bundles/paikkatietoikkuna/coordinatetransformation/components/table.js

```diff
@@ -42,45 +41,93 @@ Oskari.clazz.define('Oskari.coordinatetransformation.component.table', function(
+         * @desc handle hiding and showing the elevation row in the table
+         */
+        handleDisplayingElevationRows: function ( display ) {
+            var me = this;
+            var isEmpty = true;
+            var elevationCells = me.getElements().rows.find('.elevation');
+            if ( !display ) {
+                elevationCells.attr("contenteditable", false);
+                //check if elevationcells have value, if true don't hide but grey out
+                elevationCells.each( function (key, val) {
+                    var element = jQuery( val );
+                    element.html().trim();
+                    if ( !element.is(':empty') ) {
+                        isEmpty = false;
+                    }
+                });
+                if ( !isEmpty ) {
+                    elevationCells.addClass('cell-disabled');
+                } else {
+                    elevationCells.addClass('oskari-hidden');
+                }
+            } else {
+                elevationCells.attr("contenteditable", true);
+
+                if ( elevationCells.hasClass('cell-disabled') ) {
+                    elevationCells.removeClass('cell-disabled');
+                } else if( !elevationCells.hasClass('oskari-hidden') ) {
+                    return;
+                } else {
+                    elevationCells.removeClass('oskari-hidden');
+                }
+            }
         },
         create: function () {
             var me = this;
                 var rowcounter = this.template.rowcounter({ rows: this.loc.utils.rows })
                                                                                 
-                var coordinatefield = this.template.table({  input: this.loc.coordinatefield.input,
-                                                                north:this.loc.coordinatefield.north,
-                                                                east:this.loc.coordinatefield.east,
-                                                                ellipse_height: "" });
+                var coordinatefield = this.template.table({
+                    input: this.loc.coordinatefield.input,
+                    north:this.loc.coordinatefield.north,
+                    east:this.loc.coordinatefield.east,
+                    ellipse_elevation: "" 
+                });
 
                 var table = this.template.tableWrapper();
                 table = jQuery(table);
-
                 table.append(coordinatefield);                
-
                 table.find('.coordinatefield-table h5').append(rowcounter);
 
                 this.setElement( table );
 
+                this.bindRowCountListener();
+
                 var coords = {};
                 for( var i = 0; i < 10; i++ ) {
                     table.find("#oskari-coordinate-table").append(this.template.row( { coords: coords } ) );
```

Comment:
Did I read correctly: "table" is a container where you insert "coordinatefield" that contains the actual table? What is the magic number 10? You could find reference to the table outside the loop and only append the rows in the loop.